### ISSUE TITLE
The datafields on AutoTraitorComponent are broken

### ISSUE THREAD

**Issue by Flareguy** [space-wizards/space-station-14#28093]
## Description
The datafields on AutoTraitorComponent are non-functional.
This means you cannot remove their uplink or objectives.

**Reproduction**
1. Spawn a humanoid prototype using SyndicateFootsoldierGear with the following component attached:
```
    - type: AutoTraitor
      giveUplink: false
      giveObjectives: false
```
2. Possess them
3. Realize you still have traitor objectives
4. Use your provided uplink code, realize that you still have an uplink

**Screenshots**
![image](https://github.com/space-wizards/space-station-14/assets/78941145/7629ae68-7dd2-4f29-8897-aeecda7a09bd)
![image](https://github.com/space-wizards/space-station-14/assets/78941145/28b03742-eb15-4a93-a05a-eb5f4c1d7bfc)

**Additional context**
Possibly due to the traitor refactor. I cannot come to this conclusion to myself, since I do not know C#, but a member of the discord made this observation:
![image](https://github.com/space-wizards/space-station-14/assets/78941145/22a0360b-af0a-42bb-9fbc-10f1715d6594)

**Comment by metalgearsloth** [comment]
Isn't this code deprecated?

**Comment by Dutch-VanDerLinde** [comment]
> Isn't this code deprecated?

used on syndicate reinforcements

**Comment by metalgearsloth** [comment]
That doesn't mean it's not deprecated. Hardcoding a component just to add 1 antag seems like a code-smell when it should be able to setup any antag.

**Comment by deltanedas** [comment]
its specifically for controlling uplink and objectives

**Comment by slarticodefast** [comment]
Fixed by #30359
